dh_signobs: fix parsing of pesign hashing output

bluca opened the pull request:
At some point the output format changed and includes the filename, so the comparison of old hash vs new hash doesn't work as the filenames are included, e.g.:

$ pesign -n sql:$PWD/nss-db -h -i foo
0644c976190610760eab559bc3c3f9ef7d1541a619c9a5f4ee209da19a6d3124 foo

Fix the parsing accordingly

bluca committed:
dh_signobs: fix parsing of pesign hashing output

At some point the output format changed and includes the filename, so the comparison of
old hash vs new hash doesn't work as the filenames are included, e.g.:

$ pesign -n sql:$PWD/nss-db -h -i foo
0644c976190610760eab559bc3c3f9ef7d1541a619c9a5f4ee209da19a6d3124 foo

Fix the parsing accordingly

bluca commented:
@dirkmueller gentle ping, would appreciate a review - this fixes an issue with signing deb packages on OBS. Thanks in advance!

dirkmueller reviewed (approved):
@mlschroe any concerns?

bluca committed:
dh_signobs: fix json input for pre-configured template

The json format was changed some years ago, but never noticed with this helper
as it was usually used with autodiscovery and autogeneration instead of a pre
defined template. Tested now with a pre-defined template for systemd-boot, and
found this issue. Fix it.

bluca committed:
debian: add Provides dh-sequence-signobs

Modern debhelper tools can activate an addon via a sequence provider,
without the need to edit debian/rules to add dh --with <addon>, which is
nicer and more declarative
